### ISSUE TITLE
[v3] Implement new Concept Exercise: function-composition

### ISSUE THREAD

**Issue by ErikSchierboom** [exercism/fsharp#853]
This issue describes how to implement the `function-composition` concept exercise for the F# track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student more advanced aspects of the Concept of Composition in F#.

## Learning objectives

- Know what function associativity is.
- Know how to compose functions.
- Know what partial application is and how to use it.
- Know what currying is and how it is used in F#.
- Know how to use the pipeline operator.

## Out of scope

- Recursion.
- Computation expressions.

## Concepts

The Concepts this exercise unlocks are:

- `function-composition`: know what function associativity is; know how to compose functions; know what partial application is and how to use it; know what currying is and how it is used in F#; know how to use the pipeline operator.

## Prerequisites

This exercise's prerequisites Concepts are:

- `higher-order-functions`: know how to use higher-order functions.

Any data types used in this exercise (e.g. `strings` or `booleans`) should also be added as prerequisites.

### Hints

- [Composition and pipelining][composition-and-pipelining]: introduction to composition and pipelining.
- [Partial application][partial-application]: introduction to partial application.

### After

- [Composition and pipelining][composition-and-pipelining]: introduction to composition and pipelining.
- [Partial application][partial-application]: introduction to partial application.
- [Composition in-depth][composition-in-depth]: in-depth explanation of composition.
- [Partial application-in-depth][partial-application-in-depth]: in-depth explanation of partial application.
- [Currying in-depth][currying-in-depth]: in-depth explanation of currying.

## Representer

This exercise does not require any specific representation logic to be added to the [representer][representer].

## Analyzer

This exercise does not require any specific logic to be added to the [analyzer][analyzer].

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/fsharp/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[how-to-implement-a-concept-exercise]: https://github.com/exercism/v3/blob/master/docs/maintainers/generic-how-to-implement-a-concept-exercise.md
[implemented-exercises]: https://github.com/exercism/v3/tree/master/languages/fsharp/exercises/concept/README.md#implemented-exercises
[reference]: https://github.com/exercism/v3/blob/master/reference/concepts/composition.md
[analyzer]: https://github.com/exercism/fsharp-analyzer
[representer]: https://github.com/exercism/fsharp-representer
[meta-design]: https://github.com/exercism/v3/blob/master/languages/fsharp/exercises/concept/numbers/.meta/design.md
[meta-config.json]: https://github.com/exercism/v3/blob/master/languages/fsharp/exercises/concept/numbers/.meta/config.json
[docs-concept-exercises]: https://github.com/exercism/v3/blob/master/docs/concept-exercises.md
[composition-and-pipelining]: https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/functions/#function-composition-and-pipelining
[partial-application]: https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/functions/#partial-application-of-arguments
[composition-in-depth]: https://fsharpforfunandprofit.com/posts/function-composition/
[partial-application-in-depth]: https://fsharpforfunandprofit.com/posts/convenience-partial-application/
[currying-in-depth]: https://fsharpforfunandprofit.com/posts/currying/
